### PR TITLE
Prepare release 0.100.0

### LINKED ISSUE #2929: Release the operator v0.100.0

### PR DESCRIPTION
**Pull request by VineethReddy02**
resolves #2929

### CODE REVIEW

**Comment by pavolloffay** [comment]
@VineethReddy02 could you please update the changelog. This PR was merged https://github.com/open-telemetry/opentelemetry-operator/pull/2938/files#diff-ad5afc4cd0359615bbf52013394a16ff12b136d4e62066e576dd0c0295b2ea22R1

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,43 @@
...
+  When updating `OpenTelemetryCollector.spec.nodeSelector` it was not removing previous selector from the final collector pod (Deployment/Daemonset/Statefulset).
+  
+- `collector`: Fix of Labels and Annotations filter (#2770)
+- `target allocator`: Fix target allocator readiness check (#2903)
+
+### Components
+
+* [OpenTelemetry Collector - v0.100.0](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.100.0)
+* [OpenTelemetry Contrib - v0.100.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.100.0)
+* [Java auto-instrumentation - v1.32.0](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.32.0)
```
Comment (pavolloffay) [inline CHANGELOG.md:34]: this should be 1.32.1

**Review by frzifus — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (pavolloffay) [reply CHANGELOG.md:53]: The version still hasn't been updated

File: RELEASE.md [hunk 2]
```diff
@@ -44,9 +44,9 @@ The operator should be released within a week after the [OpenTelemetry collector
 
 | Version  | Release manager |
 |----------|-----------------|
-| v0.100.0 | @TylerHelmuth   |
```
[the remark this answers is not in the document]
Comment (TylerHelmuth) [reply RELEASE.md:47]: So close lol

**Comment by pavolloffay** [comment]
I would like to block this until https://github.com/open-telemetry/opentelemetry-operator/pull/2878 is merged

**Comment by pavolloffay** [comment]
Changelog for this can be added as well https://github.com/open-telemetry/opentelemetry-operator/pull/2964#pullrequestreview-2063141378

**Comment by VineethReddy02** [comment]
Okay, I see we are waiting for #2878 (I will update chlog once its merged)

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**